Documentation/ApiOverview/Logging/Writers/Index.rst

```diff
@@ -172,7 +172,9 @@ the :ref:`PhpErrorLog <logging-writers-php>` writer).
 Usage in Custom Class
 ---------------------
 
-All log writers can be used in your own classes. You can initialize the loggers in the __construct method of your class :code:`\MyDomain\MyExtension\MyFolder\MyClass`.
+All log writers can be used in your own classes. If the service is configured to use autowiring 
+you can initialize the loggers in the __construct method of your class :code:`\MyDomain\MyExtension\MyFolder\MyClass`) . 
+If no autowiring is used the service class however must implement the interface :code:`\Psr\Log\LoggerAwareInterface` and :code:`\Psr\Log\LoggerAwareTrait` and use no constructor.
```

Comment:
I have added a second example.

Reply:
> If no autowiring is used the service class however must implement the interface :code:`\Psr\Log\LoggerAwareInterface`, `use` the :code:`\Psr\Log\LoggerAwareTrait` and never use a constructor.

But there is an example in TYPO3 12 using TRAIT and a constructor.

    class ArrayAdapter implements AdapterInterface, CacheInterface, LoggerAwareInterface, ResettableInterface
    {
        use LoggerAwareTrait;
    
        /**
         * @param bool $storeSerialized Disabling serialization can lead to cache corruptions when storing mutable values but increases performance otherwise
         */
        public function __construct(int $defaultLifetime = 0, bool $storeSerialized = true, float $maxLifetime = 0, int $maxItems = 0)
        {

Reply:
Currently I can't test it.
So if you write about an issue you should setup perhaps an instance to test.

Reply:
The ArrayAdapter is part of symfony/cache component. So, Symfony handles it somehow. symfony/cache is not used by TYPO3, but defined as dependency from another dependency, therefore installed.

TYPO3 has a LoggerAwarePass which takes care to call the setLogger() method of the trait if the class implements the LoggerAwareInterface via DI.

Additionally, GeneralUtility::makeInstance() checks for the interface and calls then the setLogger() method.

From my POV:
- Instantiating a class with GeneralUtility::makeInstance() works in a case when scalar values are given via constructor (when also given to makeInstance()).
- DI works with static values in the constructor if they are given from Services.[yaml|php].

@DavidBruchmann Using LoggerAwareInterface and a constructor (for injecting other dependencies) works in v10 and v11, as I used them in extensions/projects. As mentioned above, it is just a compiler pass which does the magic, independent, of a constructor is available or not.